src/Experimental/RAGTools/retrieval.jl

```diff
@@ -262,6 +262,67 @@ function find_closest(
     end
 end
 
+"""
+    find_closest(
+        finder::AbstractSimilarityFinder, index::PineconeIndex,
+        query_emb::AbstractVector{<:Real}, query_tokens::AbstractVector{<:AbstractString} = String[];
+        top_k::Int = 10, kwargs...)
+
+Finds the indices of chunks that are closest to query embedding (`query_emb`) by querying Pinecone.
+
+Returns only `top_k` closest indices.
+"""
+function find_closest(
+        finder::AbstractSimilarityFinder, index::PineconeIndex,
+        query_emb::AbstractVector{<:Real}, query_tokens::AbstractVector{<:AbstractString} = String[];
+        top_n::Int = 10, kwargs...)
+    # get Pinecone info
+    pinecone_context = index.pinecone_context
+    pinecone_index = index.pinecone_index
+    pinecone_namespace = index.pinecone_namespace
+
+    # query candidates
+    pinecone_results = Pinecone.query(pinecone_context, pinecone_index,
+        Vector{Float32}(query_emb), top_n, pinecone_namespace, false, true)
+    pinecone_results_json = JSON3.read(pinecone_results)
+    matches = pinecone_results_json.matches
+
+    # get the chunks / metadata / sources / scores
+    positions = [1 for _ in matches]  # TODO: change this
+    scores = [m.score for m in matches]
```

Comment:
are the returned chunks sorted by descending score/quality? The candidate chunks tend to be sorted already